graspy/inference/latent_distribution_test.py

```diff
@@ -14,121 +14,158 @@
+from ..embed import select_dimension, AdjacencySpectralEmbed
 from ..utils import import_graph, is_symmetric
 from .base import BaseInference
+from sklearn.metrics import pairwise_distances
+from sklearn.metrics.pairwise import PAIRED_DISTANCES
+from hyppo.ksample import KSample
+
+_VALID_METRICS = list(PAIRED_DISTANCES.keys())
+_VALID_METRICS.append("gaussian")  # we have a gaussian kernel implemented too
+_VALID_TESTS = ["cca", "dcorr", "hhg", "rv", "hsic", "mgc"]
 
 
 class LatentDistributionTest(BaseInference):
     """
-    Two-sample hypothesis test for the problem of determining whether two random 
+    Two-sample hypothesis test for the problem of determining whether two random
     dot product graphs have the same distributions of latent positions.
-    
+
     This test can operate on two graphs where there is no known matching between
-    the vertices of the two graphs, or even when the number of vertices is different. 
+    the vertices of the two graphs.
     Currently, testing is only supported for undirected graphs.
 
     Read more in the :ref:`tutorials <inference_tutorials>`
 
     Parameters
     ----------
+    test : str
+        Backend hypothesis test to use, one of ["cca", "dcorr", "hhg", "rv", "hsic", "mgc"].
+        These tests are typically used for independence testing, but here they are used 
+        for a two-sample hypothesis test on the latent positions of two graphs. See 
+        :class:`hyppo.ksample.KSample` for more information. 
+
+    metric : str or function, (default="euclidean")
+        Distance metric to use, either a callable or a valid string.
+        The callable should behave similarly to :func:`sklearn.metrics.pairwise_distances`,
+        if a string should be one of the keys in `sklearn.metrics.pairwise.PAIRED_DISTANCES`
+        or "gaussian" which will use a Gaussian kernel on Euclidean distances.
+
     n_components : int or None, optional (default=None)
         Number of embedding dimensions. If None, the optimal embedding
-        dimensions are found by the Zhu and Godsi algorithm.
+        dimensions are found by the Zhu and Godsi algorithm. 
+        See :func:`~graspy.embed.selectSVD` for more information.
 
     n_bootstraps : int (default=200)
-        Number of bootstrap iterations.
+        Number of bootstrap iterations for the backend hypothesis test. See
+        :class:`hyppo.ksample.KSample` for more information.
 
-    bandwidth : float, optional (default=0.5)
-        Bandwidth to use for gaussian kernel. If None,
-        the median heuristic will be used.
+    workers : int, optional (default=1)
+        Number of workers to use. If more than 1, parallelizes the code.
 
     Attributes
     ----------
     sample_T_statistic_ : float
-        The observed difference between the embedded latent positions of the two 
+        The observed difference between the embedded latent positions of the two
         input graphs.
 
-    p_ : float
+    p_value_ : float
```

Comment:
inconsistent with semipar now.

Reply:
p_value is better. but don't want to make any changes to semipar to hold up this PR, maybe make an issue?

Reply:
#356 . resolving this.